socialhome/streams/app/tests/components/AuthorBar.tests.js

```diff
@@ -0,0 +1,210 @@
+                    console.error.calledWith("Not logged in").should.be.true
+                    console.error.restore()
+                }))
+                target.instance().unfollow()
+            })
+        })
+
+        describe("follow", () => {
+            it("should display an error message if the user is not logged in", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: null})
+                let target = mount(AuthorBar, {propsData})
+                sinon.spy(console, "error")
+                target.instance().follow()
+                console.error.calledWith("Not logged in").should.be.true
+                console.error.restore()
+            })
+
+            it("should not send an HTTP request if the user is not logged in", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: null})
+                let target = mount(AuthorBar, {propsData})
+                sinon.spy(target.instance().$http, "post")
+                target.instance().follow()
+                target.instance().$http.post.called.should.be.false
+            })
+
+            it("should send an HTTP request with the right parameters when user is logged in", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: "26", guid: "42"})
+                let target = mount(AuthorBar, {propsData})
+                sinon.spy(target.instance().$http, "post")
+                target.instance().follow()
+                target.instance().$http.post
+                    .calledWith("/api/profiles/26/add_follower/", {guid: "42"}).should.be.true
+            })
+
+            it("should show that user is following author when the HTTP request succeeds", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: "26", guid: "42", isUserFollowingAuthor: false})
+                let target = mount(AuthorBar, {propsData})
+                sinon.stub(target.instance().$http, "post").returns(new Promise(resolve => {
+                    resolve()
+                    target.instance().$data._isUserFollowingAuthor.should.be.true
+                    target.find(".unfollow-btn").length.should.equal(1)
+                }))
+                target.instance().follow()
+            })
+
+            it("should show an error message when the HTTP request fails", () => {
+                let propsData = getPropsData({currentBrowsingProfileId: "26", guid: "42", isUserFollowingAuthor: true})
+                let target = mount(AuthorBar, {propsData})
+                sinon.stub(target.instance().$http, "post").returns(new Promise(_, reject => {
+                    reject()
+                    sinon.spy(console, "error")
+                    console.error.calledWith("Not logged in").should.be.true
+                    console.error.restore()
+                }))
+                target.instance().follow()
+            })
+        })
+    })
+
+    describe("lifcycle", () => {
```

Comment:
Typo, should be `lifecycle`? Also to be honest, the extra `describe` here is probably unnecessary, since it has another `describe` child.